pkg/sql/parser/sql.y

```diff
@@ -5137,9 +5137,13 @@ show_zone_stmt:
     $$.val = &tree.ShowZoneConfig{}
   }
 
-for_or_from:
+for_with_implicit_from_alias:
   FOR
-| FROM
+| FROM { /* SKIP DOC */ }
+
+from_with_implicit_for_alias:
```

Comment:
this isn't used yet right? in that case we should probably skip it

Reply:
Oh I'm planning on using it in my next PR fixing the rest of the FOR/FROM aliases. Is it ok if I keep it in this commit?

Reply:
oh, we're going both ways? for which examples?

Reply:
(in this case i'd rather introduce it in the same commit!)

Reply:
Oh I thought that's what we wanted to do from #57638 but I'll double check.

Reply:
I don't think so.

But you can merge this by deleting those lines ;) !

Reply:
FYI based on the discussion in #57638, the statement will be changed to have FROM be the documented version but with FOR retained as an implicit alias.